shop/src/main/java/ro/msg/learning/shop/utils/mapper/OrderMapper.java

```diff
@@ -0,0 +1,23 @@
+package ro.msg.learning.shop.utils.mapper;
+
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.Order;
+
+import java.time.LocalDate;
+
+public class OrderMapper {
+    public static Order DtoToOrder(OrderDto orderDto) {
+        if (orderDto == null)
+            return null;
+        Location location = new Location(orderDto.getName(), orderDto.getCountry(), orderDto.getCity(),
+                orderDto.getCounty(), orderDto.getStreetAddress());
+        location.setId(orderDto.getLocationId());
+        Customer customer = new Customer(orderDto.getCustomer().getFirstName(), orderDto.getCustomer().getLastName(),
+                orderDto.getCustomer().getUsername(), orderDto.getCustomer().getEmailAddress());
+        customer.setId(orderDto.getCustomer().getId());
+        return new Order(location, customer, LocalDate.now(), orderDto.getCountry(), orderDto.getCity(),
```

Comment:
Same here, pass the orderDto as a parameter and use its attributes inside the constructor. It's recommended usually to have a maximum number of parameters of 3 but usually we increase it to let's say 5

Reply:
here i made extensive changes :) i hope they are fair enough. i shall see:) but just in case the mapper looks ten times simpler.